src/providers/BackgroundReplacementProvider/index.tsx

```diff
@@ -0,0 +1,186 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I created a separate provider to avoid having one big provider with so many Props and having to load and manage a lot of processors and such. This seemed a little easier for the builder in the case they just want to use blur or replacement. Can this be considered for V3?

Reply:
oh I meant a shared provider's implementation but not giving builders one provider for all filtering. Builders will have the same experience as offered in the current revision.

For example, in the Chime SDK for JavaScript, David (@davcolle) puts a shared logic into a base class or interface. Builders use the blur and replacement processors via individual APIs.

- Base: [BackgroundFilterVideoFrameProcessor](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundfilter/BackgroundFilterVideoFrameProcessor.ts)
  - Sub: [BackgroundReplacementVideoFrameProcessor](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessor.ts)
  - Sub: [BackgroundBlurVideoFrameProcessor](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts)
- Base: [BackgroundFilterVideoFrameProcessorObserver](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundfilter/BackgroundFilterVideoFrameProcessorObserver.ts)
  - Sub: [BackgroundBlurVideoFrameProcessorObserver](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessorObserver.ts)
  - Sub: [BackgroundReplacementVideoFrameProcessorObserver](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessorObserver.ts)
- Base: [BackgroundFilterVideoFrameProcessorDelegate](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundfilter/BackgroundFilterVideoFrameProcessorDelegate.ts)
  - Sub: [BackgroundBlurVideoFrameProcessorDelegate](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessorDelegate.ts)
  - Sub: [BackgroundReplacementVideoFrameProcessorDelegate](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/backgroundreplacementprocessor/BackgroundReplacementVideoFrameProcessorDelegate.ts)